external/dhcpc/dhcpc_lwip.c

```diff
@@ -111,68 +110,57 @@
  ****************************************************************************/
 int dhcp_client_start(const char *intf)
 {
-	ndbg("[DHCPC] LWIP DHCPC started\n");
-	struct netif *cur_netif;
-	cur_netif = netif_find(intf);
-	if (cur_netif == NULL) {
-		ndbg("[DHCPC] No network interface for dhcpc\n");
-		return -1;
+	int ret = -1;
+	struct req_lwip_data req;
+
+	int sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sockfd < 0) {
+		printf("socket() failed with errno: %d\n", errno);
+		return ret;
 	}
-	int32_t timeleft = CONFIG_LWIP_DHCPC_TIMEOUT;
-	struct in_addr local_ipaddr;
-	struct in_addr local_netmask;
-	struct in_addr local_gateway;
 
-	struct in_addr ip_check;
+	memset(&req, 0, sizeof(req));
+	req.type = DHCPCSTART;
+	req.host_name = intf;
 
-	/* Initialize dhcp structure if exists */
-	if (netif_dhcp_data(cur_netif)) {
-		netifapi_dhcp_stop(cur_netif);
+	ret = ioctl(sockfd, SIOCLWIP, (unsigned long)&req);
```

Comment:
I think dhcp is not related to lwip because it is upper layer of socket, ip is lower of socket right?
So, socket should decide it or ip layer, at lease not here.

Reply:
DHCP is definitely the application. However, if an user wants to run DHCPC or DHCPD implemented in LwIP, then those are running over the kernel. Thanks for the comment.